src/clients/scopes/AddScopeDialog.tsx

```diff
@@ -50,55 +100,207 @@ export const AddScopeDialog = ({
     toggleDialog();
   };
 
+  const onFilterTypeDropdownToggle = () => {
+    setIsFilterTypeDropdownOpen(!isFilterTypeDropdownOpen);
+  };
+
+  const onProtocolTypeDropdownToggle = () => {
+    setIsProtocolTypeDropdownOpen(!isProtocolTypeDropdownOpen);
+  };
+
+  const onFilterTypeDropdownSelect = (filterType: string) => {
+    if (filterType === "Name") {
+      setFilterType(FilterType.Protocol);
+    }
+    if (filterType === "Protocol") {
+      setFilterType(FilterType.Name);
+    }
+    setIsFilterTypeDropdownOpen(!isFilterTypeDropdownOpen);
+  };
```

Comment:
Here you can also compare against the enum values directly instead of string literals.
```suggestion
  const onFilterTypeDropdownSelect = (filterType: string) => {
    if (filterType === FilterType.Name) {
      setFilterType(FilterType.Protocol);
    } else if (filterType === FilterType.Protocol) {
      setFilterType(FilterType.Name);
    }

    setIsFilterTypeDropdownOpen(!isFilterTypeDropdownOpen);
  };
```